### ISSUE TITLE
0.18.0 Invalid nonce on NH

### ISSUE THREAD

**Issue by Quake4** [ethereum-mining/ethminer#1916]
**Describe the bug**
Invalid nonce on NH

**To Reproduce**
run NH mining

**Screenshots (Optional)**
![image](https://user-images.githubusercontent.com/29517243/61784511-7eb5d180-ae12-11e9-986a-b14523ddc9d5.png)

**Environment (please complete the following information):**
- Operating System: win10
- Hardware rx 584
- Ethminer Version 0.18.0
- Ethminer options used: ethminer.exe -P stratum2+tcp://xxx.Work:x@daggerhashimoto.usa.nicehash.com:3353 --api-bind 127.0.0.1:-3360 --display-interval 60 -G

**Comment by bmatthewshea** [comment]
Same issue.
Verbose output on bad nonce:
https://pastebin.com/LWpYqrQv

nVidia 1060 - ethminer 18.0 - windows10 - cuda10

command:
`ethminer.exe -v 3 --cuda --farm-recheck 1300 --HWMON 1 --api-port 3333 -P stratum2+tcp://(WALLETADDR).(MYWORKER)@daggerhashimoto.usa.nicehash.com:3353`

I do occasionally get an 'accepted' - but mostly bad nonce and also getting many disconnects from nicehash side:

```
...
 m 10:59:06 <unknown> 0:06 A2:R2 23.20 Mh - cu0 23.20 58C 50% A2:R2
cu 10:59:08 cuda-0   Job: aa8052e3... Sol: 0x27b0890006196310
 i 10:59:08 <unknown>  >> {"id":40,"method":"mining.submit","params":["3CAxwke9QXkAWkXrxLWK2tCoUpb4LT7BVw.99","00000154280a12b8","0006196310"]}
 i 10:59:08 <unknown>  << {"id":40,"result":true,"error":null}
 i 10:59:08 <unknown> **Accepted  99 ms. daggerhashimoto.usa.nicehash.com [172.65.194.252:3353]
 m 10:59:11 <unknown> 0:07 A3:R2 23.17 Mh - cu0 23.17 58C 50% A3:R2

...

cu 11:05:46 cuda-0   Job: 60694863... Sol: 0x17f97000528b6b93
 i 11:05:46 <unknown>  >> {"id":40,"method":"mining.submit","params":["3CAxwke9QXkAWkXrxLWK2tCoUpb4LT7BVw.99","00000154281f05d4","00528b6b93"]}
 X 11:05:46 <unknown> Socket read failed: An existing connection was forcibly closed by the remote host
 i 11:05:46 <unknown> Disconnected from daggerhashimoto.usa.nicehash.com [172.65.194.252:3353]
 i 11:05:46 <unknown> No connection. Suspend mining ...
 i 11:05:46 <unknown> Selected pool daggerhashimoto.usa.nicehash.com:3353
 i 11:05:46 <unknown>  >> {"id":1,"method":"mining.subscribe","params":["ethminer-0.18.0","EthereumStratum/1.0.0"]}
 i 11:05:46 <unknown>  << {"id":1,"error":null,"result":[["mining.notify","0fa517bb63e621ea1ba6efa090678a45","EthereumStratum/1.0.0"],"27accd"]}
 i 11:05:46 <unknown> Stratum mode : EthereumStratum/1.0.0 (NiceHash)
 i 11:05:46 <unknown> Established connection to daggerhashimoto.usa.nicehash.com [172.65.194.252:3353]
 i 11:05:46 <unknown> Resume mining ...
```
New Dag/etc. Delays.
Why do I keep getting disconnects? My Internet is up 100%. To prove that point, I do not get these disconnects nearly as often on my AMD Card which uses same exact connection string/wallet.

**Comment by bmatthewshea** [comment]
Same issues: 
https://github.com/ethereum-mining/ethminer/issues/1858

**Comment by dsarfati** [comment]
We are seeing the same issue as well. We currently monitor the logs for an Invalid nonce message and restart ethminer.

**Comment by dhjw** [comment]
I'm getting this too, though I notice when the miner changes nonce, e.g. ` i 00:17:13 ethminer Extranonce set to 976a45` then shares are accepted.

**Comment by AndreaLanfranchi** [comment]
Solved by #1983
